packages/client/src/store/user/types.ts

```diff
@@ -0,0 +1,29 @@
+import { TUserInfo } from '../../api';
+
+export type TUserState = {
+  user: TUserInfo | null,
+  error: string | null,
+  loading: boolean,
```

Comment:
Вместо флагов удобнее завести одно строковое поле с текущим статусом:

```suggestion
  status: 'INIT' | 'FETCHING' | 'FETCH_FULFILLED' | 'FETCH_FAILED',
```

На основе этого поля можно однозначно сказать в каком состоянии мы сейчас находимся, можно прозрачно описать переходы, будет сложне ошибиться и установить неверное состояние:

```typescript
{
  loading: true
  error: 'some error'
}
```

Для дальнейшего чтения по теме: https://bespoyasov.ru/blog/fsm-to-the-rescue/